From: Etienne Buira <etienne.buira@free.fr>
Subject: gdb: lx-symbols: do not error out on monolithic build
Date: Fri, 8 Nov 2024 17:45:32 +0100

This avoids spurious message:
(gdb) lx-symbols
loading vmlinux
No source file named kernel/module/main.c.

Link: https://lkml.kernel.org/r/Zy5ALByQtpO-ddh4@Z926fQmE5jqhFMgp6
Signed-off-by: Etienne Buira <etienne.buira@free.fr>
Cc: Andrew Ballance <andrewjballance@gmail.com>
Cc: Kieran Bingham <kbingham@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 scripts/gdb/linux/modules.py |    3 +++
 scripts/gdb/linux/symbols.py |    3 +++
 2 files changed, 6 insertions(+)

--- a/scripts/gdb/linux/modules.py~gdb-lx-symbols-do-not-error-out-on-monolithic-build
+++ a/scripts/gdb/linux/modules.py
@@ -19,6 +19,9 @@ from linux import cpus, utils, lists, co
 module_type = utils.CachedType("struct module")
 
 
+def has_modules():
+    return utils.gdb_eval_or_none("modules") is not None
+
 def module_list():
     global module_type
     modules = utils.gdb_eval_or_none("modules")
--- a/scripts/gdb/linux/symbols.py~gdb-lx-symbols-do-not-error-out-on-monolithic-build
+++ a/scripts/gdb/linux/symbols.py
@@ -178,6 +178,9 @@ lx-symbols command."""
 
         self.load_all_symbols()
 
+        if not modules.has_modules():
+            return
+
         if hasattr(gdb, 'Breakpoint'):
             if self.breakpoint is not None:
                 self.breakpoint.delete()
_
